### PR TITLE
feat: Add support for defining environment variables passed to extractors

### PR DESCRIPTION
**Pull request by Arm1stice**
This PR adds an `env` field to the `extractor_action` macro. This makes it possible to define extractor actions that require static environment variables. This functionality will be required for the new Rust extractor.

### CODE REVIEW

File: kythe/extractors/extractors.bzl [hunk 1]
```diff
@@ -4,7 +4,7 @@ The extractor_action macro plumbs together a Kythe extractor to an extra_action
 with an associated action_listener.
 """
 
-def extractor_action(name, extractor, args, mnemonics, output, needs_output = False, data = [], tags = []):
+def extractor_action(name, extractor, args, mnemonics, output, needs_output = False, data = [], tags = [], env = []):
```
Comment (shahms) [inline kythe/extractors/extractors.bzl:7]: This should likely be a dict

Comment (Arm1stice) [reply kythe/extractors/extractors.bzl:7]: Updated to use a dictionary instead

File: kythe/extractors/extractors.bzl [hunk 2]
```diff
@@ -23,15 +23,17 @@ def extractor_action(name, extractor, args, mnemonics, output, needs_output = Fa
       output: The output file template (string).
       data: A list of data dependencies (optional).
       tags: A list of build tags (optional).
+      env: A dict of environment variables (optional).
     """
     xa_name = name + "_extra_action"
 
+    env_vars = ["{}={}".format(item[0], item[1]) for item in envs.items()]
```
Comment (shahms) [inline kythe/extractors/extractors.bzl:30]: s/envs/env/

File: kythe/extractors/extractors.bzl (hunk 1 above)
Comment (shahms) [inline kythe/extractors/extractors.bzl]: `env = {}`

File: kythe/extractors/extractors.bzl [hunk 3]
```diff
@@ -23,15 +23,17 @@ def extractor_action(name, extractor, args, mnemonics, output, needs_output = Fa
       output: The output file template (string).
       data: A list of data dependencies (optional).
       tags: A list of build tags (optional).
+      env: A dict of environment variables (optional).
     """
     xa_name = name + "_extra_action"
 
+    env_vars = ["{}={}".format(item[0], item[1]) for item in env.items()]
```
Comment (shahms) [inline kythe/extractors/extractors.bzl:30]: optional: `"{}={}".format(*item)`  or `"=".join(item)` also work :-)

Comment (shahms) [reply kythe/extractors/extractors.bzl:30]: (although the latter can get grabby if both elements aren't already strings)

File: kythe/extractors/extractors.bzl (hunk 1 above)
Comment (Arm1stice) [reply kythe/extractors/extractors.bzl]: Should be fixed now. That's what I get for copy/pasting from a test file

File: kythe/extractors/extractors.bzl (hunk 3 above)
Comment (Arm1stice) [reply kythe/extractors/extractors.bzl]: Interesting, didn't know about that mechanism of tuples

**Pull request merged**